js/src/components/raise-budget-recommendation-banner/banner.js

```diff
@@ -0,0 +1,208 @@
+import { Notice, Flex, FlexBlock, FlexItem } from '@wordpress/components';
+import { useEffect } from '@wordpress/element';
+import { getHistory } from '@woocommerce/navigation';
+
+/**
+ * Internal dependencies
+ */
+import { getEditCampaignUrl } from '~/utils/urls';
+import { recordGlaEvent } from '~/utils/tracks';
+import DeltaValue from '~/components/delta-value';
+import useBudgetMetrics from '~/hooks/useBudgetMetrics';
+import useAdsCampaigns from '~/hooks/useAdsCampaigns';
+import Badge from '~/components/badge';
+import AppButton from '~/components/app-button';
+import useRaiseBudgetRecommendations from '~/hooks/useRaiseBudgetRecommendations';
+import formatNumber from '~/utils/formatNumber';
+import './index.scss';
+
+const RAISE_BUDGET_RECOMMENDATION_BANNER_CONTEXT =
+	'raise_budget_recommendation_banner';
+
+/**
+ * When the banner is shown.
+ *
+ * @event gla_raise_budget_recommendation_banner_shown
+ * @property {string} context The context in which the banner is shown. Set to 'raise_budget_recommendation_banner'.
+ */
+
+/**
+ * When the "View recommendation" button is clicked.
+ *
+ * @event gla_raise_budget_recommendation_banner_view_recommendation_clicked
+ * @property {string} context The context in which the banner is shown. Set to 'raise_budget_recommendation_banner'.
+ * @property {number} campaign_id The ID of the campaign for which the recommendation is being viewed.
+ */
+
+/**
+ * When the banner is dismissed by clicking the "Dismiss" button or the close icon.
+ *
+ * @event gla_raise_budget_recommendation_dismiss_clicked
+ * @property {string} context The context in which the banner was dismissed. Set to 'raise_budget_recommendation_banner'.
+ * @property {number} campaign_id The ID of the campaign for which the banner was dismissed.
+ */
+
+/**
+ * Displays a dismissible banner prompting users to raise the budget of a campaign.
+ *
+ * The banner is shown only if:
+ * - There are enabled campaigns.
+ * - There are relevant budget increase recommendations.
+ *
+ * When dismissed, the banner will not reappear until the expiry time elapses.
+ * Clicking "View recommendation" navigates to the recommendation details page for the campaign.
+ *
+ * @fires gla_raise_budget_recommendation_banner_shown when the banner is displayed.
+ * @fires gla_raise_budget_recommendation_banner_view_recommendation_clicked when the "View recommendation" button is clicked.
+ * @fires gla_raise_budget_recommendation_dismiss_clicked when the banner is dismissed.
+ *
+ * @param {Object} props Component properties.
+ * @param {Function} props.onBannerDismissed Callback function to call when the banner is dismissed.
+ * @return {JSX.Element|null} The banner component, or null if not applicable.
+ */
+const Banner = ( { onBannerDismissed } ) => {
+	const { campaigns: recommendedCampaigns } = useRaiseBudgetRecommendations();
+	const { data: allCampaigns } = useAdsCampaigns();
+	const recommendedCampaign = recommendedCampaigns?.[ 0 ] || {};
```

Comment:
@jjgrainger We are taking the first campaign from the list of recommended campaigns. Can you confirm this is ok and that the endpoint are returning campaigns ordered by their highest budget?

Reply:
Thanks @jjgrainger . I updated the PR by finding the highest paid campaign from the list of recommended ones.